actors/builtin/miner/miner_actor.go

```diff
@@ -562,6 +613,31 @@ func (a Actor) ConfirmSectorProofsValid(rt Runtime, params *builtin.ConfirmSecto
 
 		notifyPledgeChanged(rt, big.Sub(initialPledge, newlyVestedAmount))
 	}
+
+	// This transaction should replace the one inside the loop above.
+	// Migrate state mutations into here.
+	rt.State().Transaction(&st, func() interface{} {
+		// Mark replaced sectors for on-time expiration at the end of the current proving period.
+		// It's can't be removed right now because it may yet be challenged for Window PoSt in this period,
+		// and the deadline assignments can't be changed mid-period.
```

Comment:
If it will be challenged, shouldn't the new one be challenged instead?

E.g. we update the previous as "replaced" and when it's challenged the new sector should be challenged.

It's ok if this is not the case, we will run in a setting where both sectors can be challenged (fine from theory), just trying to understand.

Reply:
The new one won't be challenged until it gets assigned to a deadline at PP end. it's not "replaced", just "license to terminate early".